daemon/daemon/k8s_watcher.go

```diff
@@ -266,6 +266,41 @@ func (d *Daemon) syncLB() {
 			continue
 		}
 
+		isSvcIPv4 := svcInfo.IP.To4() != nil
+		if isSvcIPv4 && !d.conf.IPv4Enabled {
+			log.Warningf("Received an IPv4 kubernetes service but IPv4 is"+
+				"disabled in the cilium daemon. Ignoring service %+v", svc)
+			continue
+		}
+
+		if isSvcIPv4 {
```

Comment:
You can have a single isSvcIPv4 branch and do the !d.conf.IPv4Enabled check from within